spec.emu

```diff
@@ -676,134 +765,210 @@
+          1. Set _state_.[[GeneratedColumn]] to 0.
+          1. If |MappingList| is present, then
+            1. Perform DecodeMappingsField on |MappingList| with arguments _state_, _mappings_, _names_ and _sources_.
+        </emu-alg>
+        <emu-grammar>
+          Mapping :
+            GeneratedColumn
+        </emu-grammar>
+        <emu-alg>
+          1. Perform DecodeMappingsField of |GeneratedColumn| with arguments _state_, _mappings_, _names_ and _sources_.
+          1. Let _decodedMapping_ be a new DecodedMappingRecord { [[GeneratedLine]]: _state_.[[GeneratedLine]], [[GeneratedColumn]]: _state_.[[GeneratedColumn]], [[OriginalSource]]: *null*, [[OriginalLine]]: *null*, [[OriginalColumn]]: *null*, [[Name]]: *null* }.
+          1. Append _decodedMapping_ to _mappings_.
+        </emu-alg>
+        <emu-grammar>
+          Mapping :
+            GeneratedColumn OriginalSource OriginalLine OriginalColumn Name?
+        </emu-grammar>
+        <emu-alg>
+          1. Perform DecodeMappingsField of |GeneratedColumn| with arguments _state_, _mappings_, _names_ and _sources_.
+          1. Perform DecodeMappingsField of |OriginalSource| with arguments _state_, _mappings_, _names_ and _sources_.
+          1. Perform DecodeMappingsField of |OriginalLine| with arguments _state_, _mappings_, _names_ and _sources_.
+          1. Perform DecodeMappingsField of |OriginalColumn| with arguments _state_, _mappings_, _names_ and _sources_.
+          1. Let _source_ be _sources_[_state_.[[SourceIndex]]].
+          1. If |Name| is present, then
+            1. Perform DecodeMappingsField of |Name| with arguments _state_, _mappings_, _names_ and _sources_.
+            1. Let _name_ be _names_[_state_.[[NameIndex]]].
+          1. Else, let _name_ be *null*.
+          1. Let _decodedMapping_ be a new DecodedMappingRecord { [[GeneratedLine]]: _state_.[[GeneratedLine]], [[GeneratedColumn]]: _state_.[[GeneratedColumn]], [[OriginalSource]]: _source_, [[OriginalLine]]: _state_.[[OriginalLine]], [[OriginalColumn]]: _state_.[[OriginalColumn]], [[Name]]: _name_ }.
+          1. Append _decodedMapping_ to _mappings_.
+        </emu-alg>
+        <emu-grammar>
+          GeneratedColumn :
+            Vlq
+        </emu-grammar>
+        <emu-alg>
+          1. Let _relativeColumn_ be the VLQSignedValue of |Vlq|.
+          1. Set _state_.[[GeneratedColumn]] to _state_.[[GeneratedColumn]] + _relativeColumn_.
+          1. If _state_.[[GeneratedColumn]] &lt; 0, optionally report an error.
+        </emu-alg>
+        <emu-grammar>
+          OriginalSource :
+            Vlq
+        </emu-grammar>
+        <emu-alg>
+          1. Let _relativeSourceIndex_ the VLQSignedValue of |Vlq|.
+          1. Set _state_.[[SourceIndex]] to _state_.[[SourceIndex]] + _relativeSourceIndex_.
+          1. If _state_.[[SourceIndex]] &lt; 0 or _state.[[SourceIndex]] ≥ the number of elements of _sources_, optionally report an error.
```

Comment:
Minor typo: the second `_state_` is missing the end `_`.

Reply:
Fixed :)